packages/app-bridge/src/react/usePortalNavigation.ts

```diff
@@ -0,0 +1,167 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { produce } from 'immer';
+import debounce from 'lodash-es/debounce';
+import { useCallback, useEffect, useState } from 'react';
+import { type AppBridgeTheme } from 'src/AppBridgeTheme';
+import { type EmitterEvents } from 'src/types/Emitter';
+import { type NavigationTreeClassTypes, type NavigationTree } from 'src/types/NavigationTree';
+
+type Options = {
+    /**
+     * Whether it should fetch on mount.
+     */
+    enabled?: boolean;
+};
+
+export const usePortalNavigation = (appBridge: AppBridgeTheme, options: Options = { enabled: true }) => {
+    const [navigationItems, setNavigationItems] = useState<NavigationTree>([]);
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+
+    const refetch = useCallback(async () => {
+        setIsLoading(true);
+        setNavigationItems(await appBridge.getPortalNavigation());
+        setIsLoading(false);
+    }, [appBridge]);
+
+    useEffect(() => {
+        if (options.enabled) {
+            refetch();
+        }
+    }, [options.enabled, refetch]);
+
+    useEffect(() => {
+        const handleCoverPageAction = (event: EmitterEvents['AppBridge:GuidelineCoverPage:Action']) => {
+            switch (event.action) {
```

Comment:
For all the `handle*Action` callbacks:
 - the `switch` statement is only valid for one case, the remaining possibilities are no-op. How about replacing them with `if (event.action === 'delete')` and just forgetting about the remaining two?
 - the body is almost identical for all three - could it be just one callback for cover page, document and document group?
 - could the handlers be defined outside `usePortalNavigation`, similarly to `previewItemPositions`?

Reply:
I did it to be explicit... but you are right, refactored :)